File: Implementations/PrimeCheck.cs
using System;
namespace PrimeCheck
{
    class Prime
    {         
        static void Main(string[] args)
        {
            ulong num;
            Console.WriteLine("Enter a number");
            num = Convert.ToUInt64(Console.ReadLine());
            for (ulong i = 2; i < Math.Sqrt(num) + 1; i++) 
            {
                if (num % i == 0) 
                {
                    Console.WriteLine(String.Format("{0} is not prime: {1}*{2}", num, i, num/i));
                    return;
                }
            }
            Console.WriteLine(String.Format("{0} is prime", num));
        }
    }
}

File: Implementations/PrimeCheck.vb
Imports System

Module Module1
    Sub Main()
        Console.WriteLine("Enter a number")
        Dim numStr = Console.ReadLine()
        Dim num = CType(numStr, ULong)
        For i As ULong = 2 To Math.Sqrt(num)
            If num Mod i = 0 Then
                Console.WriteLine(num & " is not prime: " & i & "*" & num/i)
                End
            End If
        Next
        Console.WriteLine(num & " is prime")
    End Sub
End Module